src/harness/util.py

```diff
@@ -25,3 +29,92 @@ def decorated_testcase(*args, **kwargs):
+                                   low_frequency, high_frequency, device):
+  """Make PPA, PAL and Device object consistent with the inputs and position 
+  the device in the PPA Polygon at some random location
+
+    Args:
+      ppa_record: (dictionary) An object containing PPA Record.
+      pal_records: (list) A list of PAL Records which has to be
+        associated with the PPA.
+      ppa_fips_code: (number) The FIPS Code of the Census Tract where 
+      the PPA belongs.
+      ppa_census_year: (number) The census year in which the census 
+      tract was defined.
+      low_frequency: (number) The Primary Low Frequency for PAL
+      high_frequency: (number) The Primary High Frequency for PAL
+      device: (dictionary) An object containing device which has to be
+      moved into the PPA Polygon.
+    Returns:
+      A tuple consisting of PPA Record, PAL Record List and Device with 
+      modified Latitude and Longitude
+  """
+
+  previous_year_date = datetime.now().replace(year=datetime.now().year - 1)
+  next_year_date = datetime.now().replace(year=datetime.now().year + 1)
+
+  for pal_record in pal_records:
+    # Change the FIPS Code and Registration Date-Year in Pal Id
+    pal_record['palId'] = '/'.join([str(ppa_fips_code) if index == 2 else
+                           '%s-%d' % ('{:02d}'.format(previous_year_date.month),
+                                      previous_year_date.year)
+                           if index == 1 else str(val) for index, val in
+                                    enumerate(pal_record['palId'].split('/'))])
+    pal_record['userId'] = device['userId']
+    # Make the date consistent in Pal Record for Registration and License
+    pal_record = _changeDateInPpaAndPal(pal_record,
+                                        previous_year_date.strftime('%Y-%m-%dT%H:%M:%SZ'),
+                                        next_year_date.strftime('%Y-%m-%dT%H:%M:%SZ'))
+    # Change License Information in Pal
+    pal_record['license']['licenseAreaIdentifier'] = str(ppa_fips_code)
+    pal_record['license']['licenseAreaExtent'] = \
+      'zone/census_tract/census/%d/%d' % (ppa_census_year, ppa_fips_code)
+    # Change Frequency Information in Pal
+    for assign in ('primaryAssignment', 'secondaryAssignment'):
+      pal_record['channelAssignment'][assign]['lowFrequency'] = low_frequency
+      pal_record['channelAssignment'][assign]['highFrequency'] = high_frequency
+  # Make the date consistent in Ppa Record
+  ppa_record = _changeDateInPpaAndPal(ppa_record,
+                                      previous_year_date.strftime('%Y-%m-%dT%H:%M:%SZ'),
+                                      next_year_date.strftime('%Y-%m-%dT%H:%M:%SZ'))
+  # Add Pal Ids into the Ppa Record
+  ppa_record['ppaInfo']['palId'] = [pal_record['palId'] for pal_record
+                                    in pal_records]
+  device['installationParam']['latitude'], device['installationParam']['longitude'] = \
```

Comment:
If you really want to do this in this function, you should probably take an array of devices to suit the general case.

That said, I think it's better to keep this function focused (make PPA and PAL records consistent) and let the test case move the devices if it so desires -- it's only one line of code anyway.